frontend/src/metabase-lib/v1/metadata/Metadata.ts

```diff
@@ -141,8 +141,20 @@ class Metadata {
   /**
    * @deprecated load data via RTK Query - useGetCardQuery
    */
-  question(cardId: CardId | undefined | null): Question | null {
-    return (cardId != null && this.questions[cardId]) || null;
+  question(cardId: CardId | string | undefined | null): Question | null {
+    if (typeof cardId === "number") {
+      return this.questions[cardId];
+    }
+
+    if (typeof cardId === "string") {
+      for (const numericId in this.questions) {
+        if (this.questions[numericId]._card?.entity_id === cardId) {
+          return this.questions[numericId];
+        }
+      }
+    }
```

Comment:
The metadata's data structure was `Record<number, Question>` essentially, therefore you have to iterate over the entire `questions` map to grab the correct question.

This sucks. We really should switch to RTK query for `loadCard`. This is mostly just a stop-gap to proof that the flow _works_ and to convince you to use RTK query.